### PR TITLE
Prepare CSVs for 0.10.0

### PR DESCRIPTION
**Pull request by DTMad**

### CODE REVIEW

File: build/prepare_operator_bundle.sh [hunk 1]
```diff
@@ -4,6 +4,10 @@ set -eu
 
 bundle_image="./config/olm/openshift/bundle-current.Dockerfile"
 
+currentVersion=$(ls -t ./config/olm/openshift | grep -E "^(\d+\.)?(\d+\.)?(\*|\d+)$" | head -n 1)
```
Comment (baichinger) [inline build/prepare_operator_bundle.sh:7]: Quotes to the rescue.

File: config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    capabilities: Deep Insights
+    categories: Monitoring,Logging & Tracing,OpenShift Optional
+    containerImage: registry.connect.redhat.com/dynatrace/dynatrace-oneagent-operator
+    repository: https://github.com/Dynatrace/dynatrace-oneagent-operator
+    support: Dynatrace
+  name: dynatrace-monitoring.v0.0.0
+  namespace: placeholder
+spec:
+  apiservicedefinitions: {}
+  customresourcedefinitions:
```
Comment (lrgar) [inline config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml:15]: I believe this gets generated and added on the generated CSVs, so no need to have it in this file.

File: config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+    type: AllNamespaces
+  keywords:
+  - monitoring
+  - dynatrace
+  - oneagent
+  links:
+  - name: Operator Deploy Guide
+    url: https://www.dynatrace.com/support/help/shortlink/openshift-deploy
+  - name: OpenShift Monitoring Info
+    url: https://www.dynatrace.com/technologies/openshift-monitoring
```
Comment (lrgar) [inline config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml:318]: Kubernetes and OpenShift have different links.

File: config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+        path: version
+        x-descriptors:
+        - urn:alm:descriptor:text
+      - description: The timestamp when the instance was last updated.
+        displayName: Last Updated
+        path: updatedTimestamp
+        x-descriptors:
+        - urn:alm:descriptor:text
+      version: v1alpha1
+  description: |
```
Comment (lrgar) [inline config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml:260]: Kubernetes and OpenShift have different descriptions.

File: config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,326 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    alm-examples: '[]'
+    capabilities: Deep Insights
+    categories: Monitoring,Logging & Tracing,OpenShift Optional
+    containerImage: registry.connect.redhat.com/dynatrace/dynatrace-oneagent-operator
```
Comment (lrgar) [inline config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml:8]: Kubernetes and OpenShift have different images.

File: config/manifests/kustomization.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+resources:
```
Comment (lrgar) [inline config/manifests/kustomization.yaml:1]: I believe this file is not used anymore.

File: config/olm/openshift/current/manifests/dynatrace-oneagent-operator_v1_serviceaccount.yaml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
```
Comment (lrgar) [inline config/olm/openshift/current/manifests/dynatrace-oneagent-operator_v1_serviceaccount.yaml:1]: Delete this whole current directory. It doesn't seem to be used at all.

File: config/olm/openshift/0.10.0/manifests/dynatrace-oneagent-operator_v1_serviceaccount.yaml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+imagePullSecrets:
+- name: redhat-connect
+- name: redhat-connect-sso
```
Comment (lrgar) [inline config/olm/openshift/0.10.0/manifests/dynatrace-oneagent-operator_v1_serviceaccount.yaml:4]: These pull secrets aren't needed in CSVs.

File: config/olm/kubernetes/0.10.0/dynatrace-monitoring.v0.10.0.clusterserviceversion.yaml [hunk 9]
```diff
@@ -0,0 +1,802 @@
...
+            "name": "oneagentapm",
+            "namespace": "dynatrace"
+          },
+          "spec": {
+            "apiUrl": "https://ENVIRONMENTID.live.dynatrace.com/api"
+          }
+        }
+      ]
+    capabilities: Deep Insights
+    categories: Monitoring,Logging & Tracing,OpenShift Optional
```
Comment (lrgar) [inline config/olm/kubernetes/0.10.0/dynatrace-monitoring.v0.10.0.clusterserviceversion.yaml:39]: > OpenShift Optional
Should this be here?

File: config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml (hunk 5 above)
Comment (DTMad) [reply config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml:8]: I know but for the base you can only provide one of them

File: config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml (hunk 2 above)
Comment (DTMad) [reply config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml:15]: Tested and it didn't get added. Therefore I'll leave it

File: config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml (hunk 4 above)
Comment (DTMad) [reply config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml:260]: In the base file I can't provide two of them

File: config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml (hunk 3 above)
Comment (DTMad) [reply config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml:318]: In the base file I can't provide two of them

File: config/manifests/kustomization.yaml (hunk 6 above)
Comment (DTMad) [reply config/manifests/kustomization.yaml:1]: It automatically gets generated when running `make bundle`

File: config/olm/kubernetes/0.10.0/dynatrace-monitoring.v0.10.0.clusterserviceversion.yaml (hunk 9 above)
Comment (DTMad) [reply config/olm/kubernetes/0.10.0/dynatrace-monitoring.v0.10.0.clusterserviceversion.yaml:39]: Looks weird.
We used to have it for other releases as well.
However removed in the base and for K8s and OCP file

File: config/olm/openshift/0.10.0/manifests/dynatrace-oneagent-operator_v1_serviceaccount.yaml (hunk 8 above)
Comment (DTMad) [reply config/olm/openshift/0.10.0/manifests/dynatrace-oneagent-operator_v1_serviceaccount.yaml:4]: Removed

File: config/olm/openshift/current/manifests/dynatrace-oneagent-operator_v1_serviceaccount.yaml (hunk 7 above)
Comment (DTMad) [reply config/olm/openshift/current/manifests/dynatrace-oneagent-operator_v1_serviceaccount.yaml:1]: Deleted the directory and the corresponding bundle file

File: config/olm/openshift/0.10.0/manifests/dynatrace-monitoring.v0.10.0.clusterserviceversion.yaml [hunk 10]
```diff
@@ -764,4 +820,5 @@ spec:
     - dynatrace-monitoring.v0.7.1
     - dynatrace-monitoring.v0.8.0
     - dynatrace-monitoring.v0.8.1
-  version: 0.9.2
+    - dynatrace-monitoring.v0.9.5
```
Comment (lrgar) [inline config/olm/openshift/0.10.0/manifests/dynatrace-monitoring.v0.10.0.clusterserviceversion.yaml:823]: Don't think this line should be here. It stops at 0.8.1 for the last old format CSV.

File: config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml (hunk 2 above)
Comment (lrgar) [reply config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml:15]: The markers come from the the API markers from the DynaKube struct. So if they aren't getting added something is broken, and there is no point updating them on the code.

File: config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml (hunk 5 above)
Comment (lrgar) [reply config/manifests/bases/dynatrace-monitoring.clusterserviceversion.yaml:8]: Two different bases then? Can be looked at later.

**Review by lrgar — APPROVED** [review]

**Pull request merged**